test/project_types/script/layers/application/connect_app_test.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require "project_types/script/test_helper"
+
+describe Script::Layers::Application::ConnectApp do
+  let(:context) { TestHelpers::FakeContext.new(root: Dir.mktmpdir) }
+  let(:script_name) { "script_name" }
+  let(:language) { "typescript" }
+  let(:extension_point_type) { "payment-methods" }
+
+  let(:script_project_repository) { TestHelpers::FakeScriptProjectRepository.new }
+
+  before do
+    Script::Layers::Infrastructure::ScriptProjectRepository.stubs(:new).returns(script_project_repository)
+    Script::Layers::Infrastructure::ScriptProjectRepository.any_instance.stubs(:get).returns(script_project)
+  end
+
+  describe ".call" do
+    subject do
+      Script::Layers::Application::ConnectApp.call(ctx: context)
+    end
+
+    describe "when script project env is valid" do
+      let(:script_project) do
+        script_project_repository.create(
+          language: language,
+          extension_point_type: extension_point_type,
+          script_name: script_name,
+          env: ShopifyCLI::Resources::EnvFile.new(
+            api_key: "api_key",
+            secret: "secret",
+            extra: { "UUID" => "uuid" }
+          )
+        )
+      end
+
+      it "should return false and not prompt" do
+        Script::Forms::RunAgainstShopifyOrg.expects(:ask).never
+        ShopifyCLI::Shopifolk.expects(:check).never
+        refute subject
+      end
+    end
+
+    describe "when script project env is not valid" do
+      def new_app(title, api_key, secret, app_type = "custom")
+        {
+          "title" => title,
+          "apiKey" => api_key,
+          "apiSecretKeys" => [{ "secret" => secret }],
+          "appType" => app_type,
+        }
+      end
+
+      def new_org(id, name)
+        {
+          "id" => id,
+          "businessName" => name,
+        }
+      end
+
+      let(:orgs) do
+        [
+          new_org(1, "business1"),
+          new_org(2, "business2"),
+        ]
+      end
+
+      let(:apps) do
+        [
+          new_app("app1", selected_api_key, selected_secret),
+          new_app("app2", selected_api_key, selected_secret),
+        ]
+      end
+
+      let(:existing_scripts) { [] }
+      let(:orgs_with_apps) do
+        orgs.map do |org|
+          org.merge({ "apps" => apps })
+        end
+      end
+      let(:selected_org) { orgs_with_apps.first }
+      let(:selected_app) { selected_org["apps"].first }
+      let(:selected_api_key) { "default_api_key" }
+      let(:selected_secret) { "default_secret" }
+      let(:selected_uuid) { nil }
```

Comment:
Do we get something out of having these in `let` blocks? From my (reviewer) perspective, they add congnitive load (where are these used? are they overridden somwhere?) without bringing much.

I'd suggest inline these values when they're needed instead.

Reply:
Ok i've added a commit where I've moved parts of this into the `before` section, is this more what you had in mind? 

I kept them as variables with descriptive names but they only get assigned right before they get used. They only get used once so you could argue that they could also just go directly into the stub as well, but I like the extra clarity.